nmigen_boards/genesys2.py

```diff
@@ -0,0 +1,390 @@
+                 Subsignal("scl", Pins("AE19", dir="io")),
+                 Subsignal("sda", Pins("AF18", dir="io")),
+                 Attrs(IOSTANDARD="LVCMOS18")),
+        Resource("oled", 0,
+                 Subsignal("data_cmd", PinsN("AC17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("reset", PinsN("AB17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("clk", Pins("AF17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("mosi", Pins("Y15", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("vdd", PinsN("AG17", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("oled_vbat", Pins("AB22", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS33"))),
+        Resource("hdmi", 0,
+                 Subsignal("tx_clk",
+                           DiffPairs(p="AA20", n="AB20", dir="o"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("tx",
+                           DiffPairs(p="AC20 AA22 AB24",
+                                     n="AC21 AA23 AC25", dir="o"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("tx_scl", Pins("AF27", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("tx_sda", Pins("AF26", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("rx_clk",
+                           DiffPairs(p="AE28", n="AF28", dir="i")),
+                 Subsignal("rx",
+                           DiffPairs(p="AJ26 AG27 AH26",
+                                     n="AK26 AG28 AH27", dir="i"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("rx_scl", Pins("AJ28", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("rx_sda", Pins("AJ29", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33"))),
+        Resource("vga", 0,
+                 Subsignal("r", Pins("AK25 AG25 AH25 AK24 AJ24", dir="o")),
+                 Subsignal("g", Pins("AJ23 AJ22 AH22 AK21 AJ21 AK23",
+                                     dir="o")),
+                 Subsignal("b", Pins("AH20 AG20 AF21 AK20 AG22", dir="o")),
+                 Subsignal("hsync", PinsN("AF20", dir="o")),
+                 Subsignal("vsync", PinsN("AG23", dir="o")),
+                 Attrs(IOSTANDARD="LVCMOS33")),
+        *SDCardResources(0, clk="R28", cmd="R29", dat0="R26", dat1="R30",
+                        dat2="P29", dat3="T30", cd="P28",
+                        attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("sd_card_reset", 0,
+                 Pins("AE24", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("usb_otg", 0,
```

Comment:
This looks like an amalgamation of some interface, ULPI perhaps, and some power control pins, and needs to be split.